### PR TITLE
(fix) URLSearchParams.toString() behaviour is different from browsers

### PR DESCRIPTION
**Pull request by Sparkenstein**
According to the discussion at #7001 this PR fixes the issue of `URLSearchParams#toString` method behaving differently than browsers/specs

### CODE REVIEW

**Comment by bartlomieju** [comment]
@nayeemrmn please review

File: cli/rt/11_url.js [hunk 1]
```diff
@@ -876,6 +900,61 @@
     );
   }
 
+  function encodeStr(str, noEscapeTable, hexTable) {
```
Comment (nayeemrmn) [inline cli/rt/11_url.js:903]: Can you call this `encodeUrlSearchParam()`? Also I don't think this should take these second and third arguments in favour of referring to them as globals.

File: cli/rt/11_url.js [hunk 2]
```diff
@@ -876,6 +900,61 @@
...
+        out += hexTable[0xc0 | (c >> 6)] + hexTable[0x80 | (c & 0x3f)];
+        continue;
+      }
+      if (c < 0xd800 || c >= 0xe000) {
+        lastPos = i + 1;
+        out += hexTable[0xe0 | (c >> 12)] +
+          hexTable[0x80 | ((c >> 6) & 0x3f)] +
+          hexTable[0x80 | (c & 0x3f)];
+        continue;
+      }
```
Comment (nayeemrmn) [inline cli/rt/11_url.js:935]: This and other uses of `hexTable` look like a re-implementation of `encodeChar()`, which percent-encodes characters. Could you use that?

**Comment by nayeemrmn** [comment]
@Sparkenstein Using the existing devices I was able to implement it like this: https://github.com/nayeemrmn/deno/commit/c10bd3fd2d4b0269fe00af9293f7eab7f8156318.

File: cli/rt/11_url.js (hunk 2 above)
Comment (Sparkenstein) [reply cli/rt/11_url.js:935]: this is great, did not know about `encodeChar`, will use this one

**Comment by Sparkenstein** [comment]
Oh yes. I thought about simplifying the method `encodeStr` but thought it's better to rely on however it was implemented in std/node. This implementation makes code much readable. thanks for the review. do let me know if you need any more changes

**Comment by Sparkenstein** [comment]
Okay. thanks!

**Pull request merged**